### PR TITLE
Blocks page load

### LINKED ISSUE #651: [Blocks] Support functions that run on page load and functions that don't have an output
2 related issues:

1. We've gotten requests from the Hugging Face team to have functions that run on page load. For example, it could be something as simple as a table that refreshes its content by loading from an external URL whenever you reload the page. 
 
Implementation wise, it might look something like this (notice there is no `inputs` here):

```python
import pandas as pd

def get_data():
   url="https://raw.githubusercontent.com/cs109/2014_data/master/countries.csv"
   return pd.read_csv(url)


with Blocks() as blocks:
   dataframe = outputs.DataFrame()
   blocks.load(fn=get_data, outputs=dataframe)
   ...

```

2. We've also had requests for functions to call an external API after the prediction has been made:
 
image

Implementation wise, it might look something like this (notice there is no `outputs` here):

```python
import requests

def get_data(data):
   url="https://raw.githubusercontent.com/cs109/2014_data/master/countries.csv"
   requests.posts(url, data=data)


with Blocks() as blocks:
   dataframe = inputs.DataFrame()
   btn = Button()
   btn.click(fn=post_data, inputs=dataframe)
   ...

```

### PR DESCRIPTION
**Pull request by FarukOzderim**
Backend part of #651

### CODE REVIEW

**Comment by FarukOzderim** [comment]
LoL, not sure why it authored Abubakar and me inside the commit :D

**Comment by abidlabs** [comment]
Besides my comment on the example, everything else looks good.

But to clarify, this only implements the backend for #651, right? We still need to implement the frontend?

**Comment by abidlabs** [comment]
cc @pngwn

File: test/test_blocks.py [hunk 1]
```diff
@@ -48,10 +46,10 @@ def test_xray(self):
                     ct_run.click(
                         ct_model, inputs=[disease, ct_scan], outputs=ct_results
                     )
+            textbox = gr.components.Textbox()
+            demo.load(lambda x: x, [], [textbox])
```
[the remark this answers is not in the document]
Comment (FarukOzderim) [reply test/test_blocks.py:50]: oh I actually did that, but later I thought no need for a function definition, but you are right :D

[the remark this answers is not in the document]
Comment (FarukOzderim) [reply test/test_blocks.py:50]: btw it was just a test to check the config, not a demo, but updating now.

**Comment by abidlabs** [comment]
LGTM

**Review by abidlabs — APPROVED** [review]

**Pull request merged**